lib/client/client.js

```diff
@@ -56,50 +56,38 @@ Client = exports.Client = function (options) {
 util.inherits(Client, EventEmitter);
 
 Client.prototype.request = function (method, uri) {
-  var options, url, self = this, res, args = Array.prototype.slice.call(arguments),
-      callback = args.pop(),
-      body     = 'object' === typeof args[args.length - 1] && !Array.isArray(args[args.length - 1]) && args.pop(),
-      auth     = this.options.get('password') ? ':' + this.options.get('password') : '/token:' + this.options.get('token'),
-      encoded  = new Buffer(this.options.get('username') + auth).toString('base64'),
-      proxy    = this.options.get('proxy'),
-      useOAuth = this.options.get('oauth'),
-      token    = this.options.get('token');
-
-  url = assembleUrl(self, uri);
-
-  if (options) { // is this ever used?
-    self.options.headers = options;
-  } else {
-    self.options.headers = {
-      'Content-Type': 'application/json',
-      'Accept':       'application/json',
-      'User-Agent':   self.userAgent
-    };
-  }
-
-  if (useOAuth) {// token is an oauth token obtained from OAuth2
-    self.options.headers.Authorization = 'Bearer ' + token;
-  } else {// token is an API token obtained from the Zendesk Settings UI
-    self.options.headers.Authorization = 'Basic ' + encoded;
-  }
-
-  self.options.uri = url;
-  self.options.method = method || 'GET';
+  let self = this;
+  let options = Object.assign({}, this.options);
+  let url;
+  let res;
+  let args = Array.prototype.slice.call(arguments);
+  let callback = args.pop();
+  let body = 'object' === typeof args[args.length - 1] && !Array.isArray(args[args.length - 1]) && args.pop();
+  let proxy = this.options.get('proxy');
+
+  options.headers = {
+    'Authorization': _auth(self),
+    'Content-Type': 'application/json',
+    'Accept':       'application/json',
+    'User-Agent':   self.userAgent
+  };
```

Comment:
Could it be worth combining `headers` in case there are custom ones set on `options`? e.g. 
```js
options.headers = Object.assign(options.headers || {}, {
  'Authorization': _auth(self),
  'Content-Type': 'application/json',
  'Accept':       'application/json',
  'User-Agent':   self.userAgent
})
```